pallets/hyperbridge_aggregations/src/benchmarking.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2024, Horizen Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::v2::*;
+use frame_support::traits::fungible::Inspect;
+use frame_system::RawOrigin;
+use hex_literal::hex;
+use ismp::host::StateMachine;
+use sp_runtime::traits::Zero;
+
+type BalanceOf<T> = <<T as pallet_ismp::Config>::Currency as Inspect<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub mod utils {
+    use super::*;
+    use frame_support::traits::fungible::Mutate;
+    use sp_runtime::traits::Bounded;
+
+    pub static DEFAULT_EMPTY_ATT: [u8; 32] =
+        hex!("290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563");
+
+    pub static TEST_CONTRACT: [u8; 20] = hex!("d9145CCE52D386f254917e481eB44e9943F39138");
+
+    /// Return a whitelisted account with enough founds to do anything.
+    pub fn funded_account<T: Config>() -> T::AccountId {
+        let caller: T::AccountId = whitelisted_caller();
+        T::Currency::set_balance(&caller, BalanceOf::<T>::max_value() / 2u32.into());
+        caller
+    }
+}
+
+#[benchmarks]
```

Comment:
Having 3 benchmarks is not needed if in your pallet you're specifying to use only the `dispatch_aggregation `one. So you have two options here:
- You build a logic to specify the correct weight to use based on the situation (something similar to what has been done [here](https://github.com/zkVerify/zkVerify/blob/main/verifiers/groth16/src/lib.rs#L96) for instance)
- Considering there isn't much difference between `dispatch_aggregation` and `dispatch_aggregation_large_timeout` and considering that most likely all the dispatch requests will have a fee attached, perhaps we can keep only the `dispatch_aggregation_large_fee` bench

Reply:
I agree with this comment, but I believe there is also another option to mention.

If we believe that `timeout` and `fee` have a non-negligible impact on the weight of the extrinsic, which might be represented by some linear regression, they should probably be analyzed appropriately by the benchmarking framework in the valid ranges, using the appropriate parameter types such as `Linear` (for example see [here](https://github.com/zkVerify/zkVerify/blob/230d38bee61ae718870de4be0f1ed296bf7c1b2a/pallets/aggregate/src/benchmarking.rs#L81)).

I would suggest against the option 1 above in this case, since there are no separate code paths which are taken by the different branches of the to-be switch.

Reply:
The only reason why I included those 2 options is because they are only 2 params from the `dispatch_aggregation` call where the values can theoretically be large. 
In practice though, I would not expect the timeout to be that value (actually from aggregation we might even put this value to 0 since that means messages never expire and that's probably what we want)

bottom line: I'll keep `dispatch_aggregation_large_fee` and remove `dispatch_aggregation_large_timeout `

Reply:
That's fine but in the pallet you are still using the dispatch_aggregation bench and not the var_fee one. I guess we can substitute it afterwards, but just want to double check your intent here